### DISCUSSION TITLE
第2节课笔记(13班)

### DISCUSSION THREAD

**Discussion by Lum1104** [InternLM/Tutorial#120] [笔记]
13班同学第2节课笔记提交到👇

**Comment by qingyin741** [comment]
https://zhuanlan.zhihu.com/p/676399409

**Reply by Lum1104** [reply]
"某些原因"可以试着连镜像解决

**Comment by BOOMO4** [comment]
https://l055jh6h54r.feishu.cn/wiki/JH2Nwnjgii1oYckSeujcZraVnKb?from=from_copylink

**Comment by zhaoyiz0606** [comment]
https://blog.csdn.net/qq_45699866/article/details/135445155

**Comment by Taixi-Feng** [comment]
https://zhuanlan.zhihu.com/p/676674760?

**Comment by qingtao12138** [comment]
第二课笔记：
1.  大模型通常指的是机器学习或人工智能领域中参数数量巨大、拥有庞大计算能力的模型，大模型的优势在于其能够捕捉和理解数据中更为复杂、抽象的特征和关系，同时表现出很强的ICL能力；
2.  InternLM 是一个开源的轻量级训练框架，训练了如InternLM-7B和InternLM-20B等模型；Lagent 是一个轻量级、开源的基于大语言模型的智能体（agent）框架，包含LLM、Action Executor和Planning等几个部分；而浦语·灵笔是基于书生·浦语大语言模型研发的视觉-语言大模型，提供出色的图文理解和创作能力；
3.  创建开发机可以根据实际需求选择配置，如智能对话Demo和Lagent智能体工具调用Demo都只需要使用1/4的A100；浦语灵笔图文理解、创作Demo则需要2/4的A100；Demo完成后可以提前释放开发机，系统会返回多余的算力点数；
4.  参考文档可以较顺利地完成所有作业，注意代码checkout到指定的版本号；
5.  使用hf_hub_download 下载文件时记得在系统环境变量中export HF_ENDPOINT=https://hf-mirror.com/ ；
6.  开发机bash终端容易断开，可以考虑使用如tmux工具，方便重新连接到工作会话；
7. 观察Lagent工具调用Demo日志输出，可以大致了解 程序流程：（加载内置的工具） -> 根据用户prompt生成思考过程 -> 解析LLM输出，拿到Action、工具类型PythonInterpreter和Action Input-> 调用python解释器获取答案，这里使用sympy进行方程求解，返回结果给LLM->LLM结合解释器工具输出生成最终答案；
8.  观察图文创作的Demo日志输出，可以了解程序大致整体流程：根据用户prompt使用大模型生成文章（repetition_penalty设置高一些更适合文章创造场景）-> 使用大模型确定适合插入图像的段落行-> 生成每个段落的关键标题->根据关键标题从网络下载相应的图片（多张）->使用大模型对每个段落的图片进行排序，选择最优的图片展示；

**Comment by 5Fbw** [comment]
笔记2
https://fifth-tower-f54.notion.site/53713a4a8e434ade9c141bc2dcdda705?pvs=4

**Comment by burningmysoul2077** [comment]
全部： https://github.com/burningmysoul2077/Notes/blob/main/Datawhale%E7%BB%84%E9%98%9F%E5%AD%A6%E4%B9%A0/%E4%B9%A6%E7%94%9F%E6%B5%A6%E8%AF%AD%E5%AE%9E%E6%88%98/2.%E4%B9%A6%E7%94%9F%E6%B5%A6%E8%AF%ADDemo.md

**Comment by SiliconSisyphus** [comment]
https://www.notion.so/60f62dcdd0a548beacc03a02969d5ac2

**Comment by Xrcb** [comment]
https://blog.csdn.net/weixin_64546620/article/details/135485280

**Comment by QingyanMu** [comment]
https://blog.csdn.net/weixin_45834583/article/details/135489431?csdn_share_tail=%7B%22type%22%3A%22blog%22%2C%22rType%22%3A%22article%22%2C%22rId%22%3A%22135489431%22%2C%22source%22%3A%22weixin_45834583%22%7D

**Reply by Lum1104** [reply]
这里应该提交的是笔记哈

**Comment by youqi111** [comment]
https://www.yuque.com/g/youqi-pyncc/if61gr/bupgp5a4svvqoxqu/collaborator/join?token=FqrZaDdox1naxDl8&source=doc_collaborator# 《趣味demo》

**Comment by Loewen-Hob** [comment]
http://t.csdnimg.cn/k2N9S
书生浦语大模型实战营第二课: 轻松玩转书生·浦语大模型趣味 Demo

**Comment by suwief** [comment]
https://zhuanlan.zhihu.com/p/677639192
